### PR TITLE
grep rebuild

### PR DESCRIPTION
**Pull request by satmandu**
- Noticed that hash for grep wasn't matching download, so rebuilt it.
- Also added `pcre` dep.

Works properly:
- [x] x86_64
- [x] i686
- [x] armv7l

### CODE REVIEW

File: packages/grep.rb [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
-        i686: 'a6a62d2c91c346041fc4408727e387e2cc415e304d2ca7d372a97256520d6f64',
-      x86_64: '0106cbf95ada69fc0205b4af06af4e46144156934555b5bd9376e4c00d61092f',
+  binary_sha256({
+    aarch64: '15c5b97f4a77e1253220a259670d674a19e470db88a0148d79dfd6b48a74ccfa',
+     armv7l: '15c5b97f4a77e1253220a259670d674a19e470db88a0148d79dfd6b48a74ccfa',
+       i686: 'c732e8f7ce7e218553628203356d031d5a022d7a718ffa7b5529344edacead9b',
+     x86_64: 'bcb9d3d7f541fbd54170b9479a48ecb0c567a0e946a6733e1817f17f813e9260'
   })
 
+  depends_on 'pcre'
```
Comment (uberhacker) [inline packages/grep.rb:28]: Any difference between `pcre` and `pcre2`?

File: packages/grep.rb [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
 
   def self.build
-    ENV['CFLAGS'] = '-flto'
-    system "./configure #{CREW_OPTIONS}\
-    --without-included-regex"
```
Comment (uberhacker) [inline packages/grep.rb:30]: Doesn't this flag mean we want to use an external package?

File: packages/grep.rb (hunk 1 above)
Comment (satmandu) [reply packages/grep.rb:28]: Entirely different packages.

File: packages/grep.rb (hunk 2 above)
Comment (satmandu) [reply packages/grep.rb:30]: [linuxfromscratch](http://www.linuxfromscratch.org/lfs/view/10.0/chapter06/grep.html) doesn't use that flag, but [arch does](https://github.com/archlinux/svntogit-packages/blob/packages/grep/trunk/PKGBUILD). Using the builtin regex seems like the more conservative option.

Comment (uberhacker) [reply packages/grep.rb:30]: Would we need the pcre dependency still?

File: packages/grep.rb (hunk 1 above)
Comment (uberhacker) [reply packages/grep.rb:28]: Does one work any better than the other and is it relevant to grep?

Comment (satmandu) [reply packages/grep.rb:28]: I can't say. It's not an option to switch as far as I can tell.

File: packages/grep.rb (hunk 2 above)
Comment (satmandu) [reply packages/grep.rb:30]: rebuilt without, using debian build options,  but no change in the dependency.

File: packages/grep.rb (hunk 1 above)
Comment (satmandu) [reply packages/grep.rb:28]: no idea, but can't get rid of it it seems.

**Pull request merged**